query/src/main/scala/filodb/query/exec/rangefn/RangeFunction.scala

```diff
@@ -161,6 +161,44 @@ trait CounterChunkedRangeFunction[R <: MutableRowReader] extends ChunkedRangeFun
                     startTime: Long, endTime: Long): Unit
 }
 
+/**
+ * A ChunkedRangeFunction for period-counters/histograms. This does not handle resets/corrections.
+ * Data is assumed to be ordered.
+ */
+trait PeriodCounterChunkedRangeFunction[R <: MutableRowReader] extends ChunkedRangeFunction[R] {
```

Comment:
I would like to understand the motivation for the new traits. I was under the impression that we could do this with existing constructs. Let us discuss in person.

Reply:
changed the implementation, reusing existing `sumovertime` function